File: exampleProjects/GoExample/src/de/sots/cellarsandcentaurs/domain/service/creatures.go
package service

import "github.com/sots/cellarsandcentaurs/src/de/sots/cellarsandcentaurs/domain/model"

type Creatures interface {
	Save(creature *model.Creature) error
	FindById(id *model.CreatureId) (*model.Creature, error)
	FindAll() ([]*model.Creature, error)
	Delete(id *model.CreatureId) error
}

File: exampleProjects/PhpExample/src/de/sots/cellarsandcentaurs/domain/service/Creatures.php
<?php
namespace de\sots\cellarsandcentaurs\domain\service;

use de\sots\cellarsandcentaurs\domain\model\Creature;
use de\sots\cellarsandcentaurs\domain\model\CreatureId;

interface Creatures
{
    public function save(Creature $creature);
    public function find(CreatureId $id): Creature;
}
